stackify/transport/__init__.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import os
+import sys
+import urllib
+
+from stackify.constants import DEFAULT_SOCKET_FILE
+from stackify.constants import LOG_SAVE_URL
+from stackify.constants import SOCKET_LOG_URL
+from stackify.constants import TRANSPORT_TYPE_AGENT_SOCKET
+from stackify.constants import TRANSPORT_TYPE_DEFAULT
+from stackify.transport.agent import AgentSocket
+from stackify.transport.agent.message import Log
+from stackify.transport.agent.message import LogGroup
+from stackify.transport.application import get_configuration
+from stackify.transport.application import EnvironmentDetail
+from stackify.transport.default import HTTPClient
+from stackify.transport.default.log import LogMsg
+from stackify.transport.default.log import LogMsgGroup
+
+PY2 = sys.version_info[0] == 2
+
+internal_logger = logging.getLogger(__name__)
+
+
+class TransportTypes(object):
+    """
+    Transport Type class that will determine which transport to use
+    depending on users config.
+
+    Types:
+    * DEFAULT - HTTP transport that will directly send logs to the Platform
+    * AGENT_SOCKET - HTTP warapped Unix Socket Domain that will send logs to the StackifyAgent
+    """
+
+    DEFAULT = TRANSPORT_TYPE_DEFAULT
+    AGENT_SOCKET = TRANSPORT_TYPE_AGENT_SOCKET
+
+    @classmethod
+    def get_transport(self, api_config=None, env_details=None):
+        if api_config.transport == self.AGENT_SOCKET:
+            socket_url = urllib.unquote_plus(DEFAULT_SOCKET_FILE) if PY2 else urllib.parse.unquote_plus(DEFAULT_SOCKET_FILE)
+            # checking if socket file exists
+            if os.path.exists(socket_url):
```

Comment:
The transport should be created-able if the socket file does not exist - the socket file may not exist at any given point. In k8s the daemonset pod may be started after the application container or may be refreshed - we want to ensure we load the transport and just retry on posts.

Reply:
will update and remove checking